src/main/java/hudson/plugins/scm/koji/client/KojiBuildDownloader.java

```diff
@@ -240,6 +260,20 @@ private String composeUrl(String kojiDownloadUrl, Build build, RPM rpm, String s
         return sb.toString();
     }
 
+    private String composeSrcUrl(String kojiDownloadUrl, Build build, String suffix){
+        StringBuilder sb = new StringBuilder(255);
+        sb.append(kojiDownloadUrl);
+        if (kojiDownloadUrl.charAt(kojiDownloadUrl.length() - 1) != '/') {
+            sb.append('/');
+        }
+        sb.append(build.getName()).append('/');
```

Comment:
you can chain this as
``` 
sb.append(build.getName()).append('/')
  .append(build.getVersion()).append('/')
  .append(build.getRelease()).append('/')
...
```
I found it more readable, but it's personal preference...